packages/synapse-react-client/src/components/DataGrid/DataGridTypes.ts

```diff
@@ -1,3 +1,24 @@
+import { Model } from 'json-joy/lib/json-crdt'
+import { s } from 'json-joy/lib/json-crdt-patch'
+
+const gridRowSchema = s.obj({
+  data: s.vec(s.con('')),
+  metadata: s.obj({
+    synapseRow: s.obj({
+      rowId: s.val(s.con(0)),
+      versionNumber: s.val(s.con(0)),
+      etag: s.val(s.con('')),
+    }),
+  }),
+})
+
+const gridSchema = s.obj({
+  doc_version: s.con('0.0.3'),
+  columnNames: s.vec(s.con('')),
+  columnOrder: s.arr([s.con(0)]),
+  rows: s.arr([gridRowSchema]),
+})
```

Comment:
Define the schema using json-joy. This will generate types that have our model built-in, giving us enhanced type safety.